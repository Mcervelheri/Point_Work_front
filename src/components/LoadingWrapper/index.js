import React, { useMemo } from 'react';

import { Spin } from 'antd';

import useClassNames from '../../hooks/use-classnames';

import styles from './index.module.scss';

const LoadingWrapper = ({
    loading,
    loadingArr = [],
    children,
    wrapperClassName,
    spinnerClassName,
}) => {

    const loadingResult = useMemo(() => {
        if (loading !== undefined) return loading;

        return loadingArr.find(l => l === true);
    }, [loading, loadingArr]);

    const wrapperClassNameFinal = useClassNames([styles.loadingWrapper, wrapperClassName]);
    const spinnerContainerClassNameFinal = useClassNames([styles.spinnerContainer, spinnerClassName]);

    return (
        <div
            className={loadingResult ? wrapperClassNameFinal : null}
        >
            {loadingResult ? (
                <div className={spinnerContainerClassNameFinal}>
                    <span className={styles.spinner}>
                        <Spin
                            role="status"
                            animation="border"
                        />
                    </span>
                </div>
            ) : null}
            {children}
        </div>
    );
};

export default LoadingWrapper;
