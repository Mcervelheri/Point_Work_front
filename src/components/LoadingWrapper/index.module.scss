.loadingWrapper {
    width: 100%;
    height: 100%;
    opacity: 0.8;
    background-color: white;
    pointer-events: none;
    position: relative;
}

.spinner {
    transform: translate(50%, -50%);
}

.spinnerContainer {
    top: 0;
    left: 0;
    position: absolute;
    z-index: 1;

    width: 100%;
    padding-top: 50px;

    display: flex;
    justify-content: center;
    align-items: center;
}
