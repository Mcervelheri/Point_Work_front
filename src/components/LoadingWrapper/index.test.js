import React from 'react';

import renderer from 'react-test-renderer';

import LoadingWrapper from './index';

test('O componente <LoadingWrapper /> deve renderiza o Spinner quando a propriedade loading for true', () => {
    const Wrapper = renderer.create(
        <LoadingWrapper
            loading
        >
            <div>
                Renderiza o spinnerContainer
            </div>
        </LoadingWrapper>,
    );

    expect(Wrapper.toJSON()).toMatchSnapshot();
});

test('O componente <LoadingWrapper /> NÃO deve renderiza o Spinner quando a propriedade loading for false', () => {
    const Wrapper = renderer.create(
        <LoadingWrapper>
            <div>
                Não renderiza o spinnerContainer
            </div>
        </LoadingWrapper>,
    );

    expect(Wrapper.toJSON()).toMatchSnapshot();
});
