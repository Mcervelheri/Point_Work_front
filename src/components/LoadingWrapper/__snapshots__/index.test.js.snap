// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`O componente <LoadingWrapper /> NÃO deve renderiza o Spinner quando a propriedade loading for false 1`] = `
<div
  className={null}
>
  <div>
    Não renderiza o spinnerContainer
  </div>
</div>
`;

exports[`O componente <LoadingWrapper /> deve renderiza o Spinner quando a propriedade loading for true 1`] = `
<div
  className="loadingWrapper"
>
  <div
    className="spinnerContainer"
  >
    <span
      className="spinner"
    >
      <div
        animation="border"
        aria-busy={true}
        aria-live="polite"
        className="ant-spin ant-spin-spinning"
        role="status"
      >
        <span
          className="ant-spin-dot ant-spin-dot-spin"
        >
          <i
            className="ant-spin-dot-item"
          />
          <i
            className="ant-spin-dot-item"
          />
          <i
            className="ant-spin-dot-item"
          />
          <i
            className="ant-spin-dot-item"
          />
        </span>
      </div>
    </span>
  </div>
  <div>
    Renderiza o spinnerContainer
  </div>
</div>
`;
